clients/python-wrapper/lakefs/object.py

```diff
@@ -676,13 +676,15 @@ def reader(self, mode: ReadModes = 'rb', pre_sign: Optional[bool] = None) -> Obj
         """
         return ObjectReader(self, mode=mode, pre_sign=pre_sign, client=self._client)
 
-    def stat(self) -> ObjectInfo:
+    def stat(self, pre_sign: Optional[bool] = None) -> ObjectInfo:
         """
         Return the Stat object representing this object
         """
-        if self._stats is None:
+        # don't cache pre-signed stats
```

Comment:
Q. did we had a requirement to add caching? it works for a single app, but any change from outside the app will make this case invalid. it is also true for committed data if we enable retention.